### PR TITLE
chore: add appup update script

### PR DESCRIPTION
**Pull request by zmstone**

### CODE REVIEW

**Review by ieQu1 — APPROVED** [review]

**Review by ieQu1 — DISMISSED** [review]

File: scripts/update_appup.escript [hunk 1]
```diff
@@ -32,8 +32,8 @@ Options:
   --make-command    A command used to assemble the release
   --release-dir     Release directory
   --src-dirs        Directories where source code is found. Defaults to '{src,apps,lib-*}/**/'
-  --binary-rel-url  Binary release URL pattern. %TAG% variable is substituted with the release tag.
-                    E.g. \"https://github.com/emqx/emqx/releases/download/v%TAG%/emqx-centos7-%TAG%-amd64.zip\"
+  --binary-rel-url  Binary release URL pattern.
+                    E.g. https://www.emqx.com/downloads/broker/v4.3.12/emqx-cnetos7-4.3.12-amd64.zip
```
Comment (ieQu1) [inline scripts/update_appup.escript:36]:
```suggestion
                    E.g. https://www.emqx.com/downloads/broker/v4.3.12/emqx-centos7-4.3.12-amd64.zip
```

**Review by ieQu1 — APPROVED** [review]

**Review by thalesmg — APPROVED** [review]

File: scripts/update-appup.sh [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+    x86_64)
+        ARCH='amd64'
+        ;;
+    aarch64)
+        ARCH='arm64'
+        ;;
+    arm*)
+        ARCH=arm
+        ;;
+esac
```
[the remark this answers is not in the document]
Comment (zmstone) [reply scripts/update-appup.sh:55]: yes, see the line above: `ARCH="${ARCH:-$(uname -m)}"`

File: scripts/update_appup.escript [hunk 3]
```diff
@@ -171,9 +171,10 @@ download_prev_release(Tag, #{binary_rel_url := {ok, URL0}, clone_url := Repo}) -
     BaseDir = "/tmp/emqx-baseline-bin/",
     Dir = filename:basename(Repo, ".git") ++ [$-|Tag],
     Filename = filename:join(BaseDir, Dir),
-    Script = "mkdir -p ${OUTFILE} &&
-              wget -c -O ${OUTFILE}.zip ${URL} &&
-              unzip -n -d ${OUTFILE} ${OUTFILE}.zip",
+    Script = "echo \"Download: ${OUTFILE}\" &&
+              mkdir -p ${OUTFILE} &&
+              curl -f -L -o ${OUTFILE}.zip ${URL} &&
```
[the remark this answers is not in the document]
Comment (zmstone) [reply scripts/update_appup.escript:176]: good point, but wget is full of surprises, i'll add a check in the next pr

File: scripts/update-appup.sh (hunk 2 above)
[the remark this answers is not in the document]
Comment (thalesmg) [reply scripts/update-appup.sh:55]: Right, but this `case` will still transform the value passed in.  For example, if `arm64` is passed in, it'll still get transformed into `arm` by this.

[the remark this answers is not in the document]
Comment (zmstone) [reply scripts/update-appup.sh:55]: you are right. i'll fix it.

**Pull request merged**